cli/src/pcluster/models/imagebuilder.py

```diff
@@ -112,62 +161,78 @@ def __init__(self, image_name: str, config: dict = None, stack: ImageBuilderStac
         self.__source_config = config
         self.__stack = stack
+        self.__image = None
         self.__config = None
         self.__bucket = None
         self.template_body = None
         self._s3_artifacts_dict = {
             "root_directory": "parallelcluster",
-            "root_image_directory": "imagebuilders",
-            "source_config_name": "image-config-original.yaml",
+            "root_image_directory": "images",
             "config_name": "image-config.yaml",
             "template_name": "aws-parallelcluster-imagebuilder.cfn.yaml",
             "custom_artifacts_name": "artifacts.zip",
         }
         self.__s3_artifact_dir = None
 
     @property
-    def s3_artifacts_dir(self):
+    def s3_artifact_dir(self):
         """Get s3 artifacts dir."""
         if self.__s3_artifact_dir is None:
-            self._get_artifact_dir()
+            self.__s3_artifact_dir = self._get_artifact_dir()
         return self.__s3_artifact_dir
 
+    @property
+    def config_url(self):
+        """Return configuration file S3 bucket url."""
+        return self.bucket.get_config_url(self._s3_artifacts_dict.get("config_name"))
+
     @property
     def stack(self):
         """Return the ImageBuilderStack object."""
         if not self.__stack:
             try:
-                self.__stack = ImageBuilderStack(AWSApi().instance().cfn.describe_stack(self.image_name))
-            except AWSClientError:
-                raise ImageBuilderActionError(f"ImageBuilder stack {self.image_name} does not exist.")
+                self.__stack = ImageBuilderStack(AWSApi.instance().cfn.describe_stack(self.image_name))
+            except AWSClientError as e:
+                if f"Stack with id {self.image_name} does not exist" in str(e):
+                    raise NonExistingStackError(self.image_name)
+                raise StackError(f"Unable to find imagebuilder stack {self.image_name}, due to {e}")
         return self.__stack
 
     @property
-    def source_config(self):
-        """Return config used to create the imagebuilder stack."""
-        if not self.__source_config:
+    def image(self):
+        """Return Image object."""
+        if not self.__image:
             try:
-                self.__source_config = self.bucket.get_config(config_name=self._s3_artifacts_dict.get("config_name"))
-            except Exception as e:
-                raise ImageBuilderActionError(
-                    "Unable to load configuration from bucket " f"'{self.bucket.name}/{self.s3_artifacts_dir}'.\n{e}"
-                )
-        return self.__source_config
+                self.__image = AWSApi.instance().ec2.describe_image_by_name_tag(self.image_name)
+            except AWSClientError as e:
+                if isinstance(e, ImageNotFoundError):
+                    raise NonExistingImageError(self.image_name)
+                raise ImageError(f"Unable to get Image {self.image_name} info, due to {e}.")
```

Comment:
I'm not sure we really need ImageError and StackError, maybe we can raise ImageBuilderActionError directly but let's talk about it.

Reply:
we can raise the ImageBuilderActionError in stack/image property rather than StackError/ImageError, but my concern is if we do that, will we lose the error trace, for example:
```
    @property
    def imagebuild_status(self):
        """Return the status of the stack of build image process."""
        try:
            cfn_status = self.stack.status
            for key, value in ImageBuilderStatusMapping.items():
                if cfn_status in value:
                    return key
            return None
        except StackError as e:
            raise ImageBuilderActionError(f"Unable to get imagebuilder {self.image_name} status, due to {e}")
```
If we raise error in self.stack, we don't know the error is caused by imagebuild_status